detox/src/android/core/NativeElement.js

```diff
@@ -42,12 +42,31 @@ class NativeElement {
     return await new ActionInteraction(this._invocationManager, this._matcher, action, traceDescription).execute();
   }
 
-  async longPress() {
-    const action = new actions.LongPressAction();
-    const traceDescription = actionDescription.longPress();
+  async longPress(arg1, arg2) {
+    let duration = null;
+    let point = null;
+
+    if (arg1 !== undefined && typeof arg1 === 'number' && arg2 === undefined) {
```

Comment:
There's so much validation logic that I'd suggest extracting it to one more validation function.

Reply:
yep